tests/UI/campaigns/regression/checkout/100PercentDiscount_FO.js

```diff
@@ -0,0 +1,417 @@
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'addProductToCart',
+        baseContext,
+      );
+
+      await homePage.addProductToCartByQuickView(page, 1, '1');
+      await homePage.proceedToCheckout(page);
+      const pageTitle = await cartPage.getPageTitle(page);
+      await expect(pageTitle).to.equal(cartPage.pageTitle);
+    });
+
+    it('should add our discount code and check that the total price is 0', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'addPercentDiscount',
+        baseContext,
+      );
+
+      await cartPage.addPromoCode(page, percentCartRule.code);
+
+      const totalPrice = await cartPage.getATIPrice(page);
+      await expect(totalPrice, 'Order total price is incorrect').to.equal(0);
+    });
+
+    it('should go to checkout process', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'proceedToCheckout',
+        baseContext,
+      );
+
+      await cartPage.clickOnProceedToCheckout(page);
+      const isCheckoutPage = await checkoutPage.isCheckoutPage(page);
+      await expect(isCheckoutPage).to.be.true;
+    });
+
+    it('should fill personal information as a guest', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'setPersonalInformation',
+        baseContext,
+      );
+
+      const isStepPersonalInfoCompleted = await checkoutPage.setGuestPersonalInformation(
+        page,
+        customerData,
+      );
+      await expect(
+        isStepPersonalInfoCompleted,
+        'Step personal information is not completed',
+      ).to.be.true;
+    });
+
+    it('should fill address form and go to delivery step', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'setAddressStep',
+        baseContext,
+      );
+
+      const isStepAddressComplete = await checkoutPage.setAddress(
+        page,
+        addressData,
+      );
+
+      await expect(isStepAddressComplete, 'Step Address is not complete')
+        .to.be.true;
+    });
+
+    it('should go to last step', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'goToLastStep',
+        baseContext,
+      );
+
+      // Delivery step - Go to payment step
+      const isStepDeliveryComplete = await checkoutPage.goToPaymentStep(page);
+      await expect(isStepDeliveryComplete, 'Step Address is not complete')
+        .to.be.true;
+    });
+
+    it('should contain no payment needed text', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'checkNoPaymentNeededText',
+        baseContext,
+      );
+
+      const noPaymentNeededText = await checkoutPage.getTextContent(
+        page,
+        checkoutPage.noPaymentNeededElement,
+      );
```

Comment:
```suggestion
      const noPaymentNeededText = await checkoutPage.getNoPaymentNeededBlockContent(page);
```

Same as first comment